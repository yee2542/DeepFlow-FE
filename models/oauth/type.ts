import { OAuthEnum } from './enum';

export type OAuthType = OAuthEnum.Facebook;
