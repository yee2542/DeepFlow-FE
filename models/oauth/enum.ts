export enum OAuthEnum {
  Facebook = 'facebook',
}
