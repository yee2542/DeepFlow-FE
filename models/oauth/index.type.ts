import { OAuthEnum } from './oauth.enum';

export type OAuthType = OAuthEnum.Facebook;
