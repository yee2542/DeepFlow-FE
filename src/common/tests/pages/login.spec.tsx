import FormCenterLayout from 'Modules/user/components/Layouts/FormCenter';
import { mount, shallow } from 'enzyme';
import { OAuthEnum } from 'Models/oauth/enum';
import React from 'react';
import LoginPage from 'Modules/user/pages/login';

describe('Pages/Login test', () => {
  it('Render correctly', () => {
    const wrap = shallow(<LoginPage />);
    expect(wrap.html()).toMatchSnapshot();
  });

  it('LoginPage should be exisiting', () => {
    const wrap = shallow(<LoginPage />);
    expect(wrap.exists()).toBe(true);
  });

  it('Should contain FormCenter Layout', () => {
    const wrap = shallow(<LoginPage />);
    expect(wrap.exists(FormCenterLayout)).toBe(true);
  });

  it('Should have email & password field / submit btn', () => {
    const wrap = mount(<LoginPage />);
    const emailField = wrap.find('#email');
    const passwordField = wrap.find('#password');
    const submitButton = wrap.find('#login-btn');

    expect(emailField.exists()).toBe(true);
    expect(passwordField.exists()).toBe(true);
    expect(submitButton.exists()).toBe(true);
  });

  it('Should call onError when fill only email', () => {
    const onError = jest.fn(() => expect(onError).toBeCalledTimes(1));
    const onSubmit = jest.fn(() => expect(onSubmit).toBeCalledTimes(0));

    const wrap = mount(<LoginPage onSubmit={onSubmit} onError={onError} />);
    const emailField = wrap.find('input[id="email"]');
    emailField.simulate('change', { target: { value: 'Hello@email.com' } });

    wrap.find('#login-btn').at(0).simulate('submit');
  });

  it('Should call onError when not a email', () => {
    const onError = jest.fn(() => expect(onError).toBeCalledTimes(1));
    const onSubmit = jest.fn(() => expect(onSubmit).toBeCalledTimes(0));

    const wrap = mount(<LoginPage onSubmit={onSubmit} onError={onError} />);
    const emailField = wrap.find('input[id="email"]');
    emailField.simulate('change', { target: { value: 'Hello' } });

    wrap.find('#login-btn').at(0).simulate('submit');
  });

  it('Should call onSubmit when fullfill', () => {
    const onSubmit = jest.fn(() => expect(onSubmit).toBeCalledTimes(1));
    const onError = jest.fn(() => expect(onError).toBeCalledTimes(0));

    const wrap = mount(<LoginPage onSubmit={onSubmit} onError={onError} />);
    const emailField = wrap.find('input[id="email"]');
    const passwordField = wrap.find('input[id="password"]');

    emailField.simulate('change', { target: { value: 'Hello' } });
    passwordField.simulate('change', { target: { value: '1234' } });

    wrap.find('#login-btn').at(0).simulate('submit');
  });

  it('Should call onSubmitOAuth and send OAuthEnum.Facebook via callback', () => {
    const onSumitOAuth = jest.fn((OAuth) => {
      expect(onSumitOAuth).toBeCalledTimes(1);
      expect(OAuth).toEqual(OAuthEnum.Facebook);
    });
    const wrap = mount(<LoginPage onSubmitOAuth={onSumitOAuth} />);

    wrap.find('#fb-btn').at(0).simulate('click');
  });
});
