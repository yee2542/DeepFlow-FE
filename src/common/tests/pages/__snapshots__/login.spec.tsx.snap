// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pages/Login test Render correctly 1`] = `"<div class=\\"styled__Container-sc-1un2s9o-0 YXrJt\\"><div class=\\"ant-row ant-row-center\\"><div class=\\"ant-col ant-col-xs-0 ant-col-sm-0 ant-col-md-0 ant-col-lg-12\\"><div style=\\"display:block;overflow:hidden;position:relative;box-sizing:border-box;margin:0\\"><div style=\\"display:block;box-sizing:border-box;padding-top:166.66666666666669%\\"></div><img src=\\"data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7\\" decoding=\\"async\\" style=\\"visibility:hidden;position:absolute;top:0;left:0;bottom:0;right:0;box-sizing:border-box;padding:0;border:none;margin:auto;display:block;width:0;height:0;min-width:100%;max-width:100%;min-height:100%;max-height:100%\\"/></div></div><div class=\\"ant-col ant-col-xs-20 ant-col-sm-20 ant-col-md-20 ant-col-lg-12\\"><div class=\\"styled__FormContainer-sc-1un2s9o-1 eFwkdR\\"><div class=\\"ant-row ant-row-center ant-row-middle\\"><div class=\\"ant-col ant-col-8\\"><div style=\\"display:inline-block;max-width:100%;overflow:hidden;position:relative;box-sizing:border-box;margin:0\\"><div style=\\"box-sizing:border-box;display:block;max-width:100%\\"><img style=\\"max-width:100%;display:block;margin:0;border:none;padding:0\\" alt=\\"\\" aria-hidden=\\"true\\" role=\\"presentation\\" src=\\"data:image/svg+xml;base64,PHN2ZyB3aWR0aD0iMjAwIiBoZWlnaHQ9IjEyMCIgeG1sbnM9Imh0dHA6Ly93d3cudzMub3JnLzIwMDAvc3ZnIiB2ZXJzaW9uPSIxLjEiLz4=\\"/></div><img src=\\"data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7\\" decoding=\\"async\\" style=\\"visibility:hidden;position:absolute;top:0;left:0;bottom:0;right:0;box-sizing:border-box;padding:0;border:none;margin:auto;display:block;width:0;height:0;min-width:100%;max-width:100%;min-height:100%;max-height:100%\\"/></div></div></div><div class=\\"ant-row ant-row-center\\"><div class=\\"ant-col ant-col-16 ant-col-xl-14\\"><form id=\\"login-form\\" class=\\"ant-form ant-form-vertical ant-form-large\\"><div class=\\"ant-row ant-form-item\\"><div class=\\"ant-col ant-col-24 ant-form-item-control\\"><div class=\\"ant-form-item-control-input\\"><div class=\\"ant-form-item-control-input-content\\"><span class=\\"ant-input-affix-wrapper ant-input-affix-wrapper-lg\\"><span class=\\"ant-input-prefix\\"><span role=\\"img\\" aria-label=\\"mail\\" class=\\"anticon anticon-mail\\"><svg viewBox=\\"64 64 896 896\\" focusable=\\"false\\" data-icon=\\"mail\\" width=\\"1em\\" height=\\"1em\\" fill=\\"currentColor\\" aria-hidden=\\"true\\"><path d=\\"M928 160H96c-17.7 0-32 14.3-32 32v640c0 17.7 14.3 32 32 32h832c17.7 0 32-14.3 32-32V192c0-17.7-14.3-32-32-32zm-40 110.8V792H136V270.8l-27.6-21.5 39.3-50.5 42.8 33.3h643.1l42.8-33.3 39.3 50.5-27.7 21.5zM833.6 232L512 482 190.4 232l-42.8-33.3-39.3 50.5 27.6 21.5 341.6 265.6a55.99 55.99 0 0068.7 0L888 270.8l27.6-21.5-39.3-50.5-42.7 33.2z\\"></path></svg></span></span><input type=\\"text\\" id=\\"email\\" placeholder=\\"email\\" value=\\"\\" class=\\"ant-input ant-input-lg\\"/></span></div></div></div></div><div class=\\"ant-row ant-form-item\\"><div class=\\"ant-col ant-col-24 ant-form-item-control\\"><div class=\\"ant-form-item-control-input\\"><div class=\\"ant-form-item-control-input-content\\"><span class=\\"ant-input-affix-wrapper ant-input-affix-wrapper-lg\\"><span class=\\"ant-input-prefix\\"><span role=\\"img\\" aria-label=\\"lock\\" class=\\"anticon anticon-lock\\"><svg viewBox=\\"64 64 896 896\\" focusable=\\"false\\" data-icon=\\"lock\\" width=\\"1em\\" height=\\"1em\\" fill=\\"currentColor\\" aria-hidden=\\"true\\"><path d=\\"M832 464h-68V240c0-70.7-57.3-128-128-128H388c-70.7 0-128 57.3-128 128v224h-68c-17.7 0-32 14.3-32 32v384c0 17.7 14.3 32 32 32h640c17.7 0 32-14.3 32-32V496c0-17.7-14.3-32-32-32zM332 240c0-30.9 25.1-56 56-56h248c30.9 0 56 25.1 56 56v224H332V240zm460 600H232V536h560v304zM484 701v53c0 4.4 3.6 8 8 8h40c4.4 0 8-3.6 8-8v-53a48.01 48.01 0 10-56 0z\\"></path></svg></span></span><input type=\\"password\\" id=\\"password\\" placeholder=\\"Password\\" value=\\"\\" class=\\"ant-input ant-input-lg\\"/></span></div></div></div></div><div class=\\"ant-row ant-form-item\\"><div class=\\"ant-col ant-col-24 ant-form-item-control\\"><div class=\\"ant-form-item-control-input\\"><div class=\\"ant-form-item-control-input-content\\"><button id=\\"login-btn\\" type=\\"submit\\" class=\\"ant-btn ant-btn-primary ant-btn-lg ant-btn-block\\"><span>Log in</span></button></div></div></div></div><div class=\\"ant-row ant-row-center\\"><div class=\\"ant-col ant-col-10\\"><hr/></div><div class=\\"ant-col ant-col-4\\"><p class=\\"global__P-m421n3-0 iUPqqF\\">or</p></div><div class=\\"ant-col ant-col-10\\"><hr/></div></div><div class=\\"ant-row ant-form-item\\"><div class=\\"ant-col ant-col-24 ant-form-item-control\\"><div class=\\"ant-form-item-control-input\\"><div class=\\"ant-form-item-control-input-content\\"><button id=\\"fb-btn\\" style=\\"background-color:#3b5998;color:#ffffff\\" type=\\"button\\" class=\\"ant-btn ant-btn-lg ant-btn-block\\"><span>Login with Facebook</span></button></div></div></div></div><div class=\\"ant-row ant-form-item\\"><div class=\\"ant-col ant-col-24 ant-form-item-control\\"><div class=\\"ant-form-item-control-input\\"><div class=\\"ant-form-item-control-input-content\\"><p class=\\"global__P-m421n3-0 iUPqqF\\">Haven&#x27;t an account?<a href=\\"/register\\"> Register</a></p></div></div></div></div></form></div></div></div></div></div></div>"`;
